tests/erc721/erc721_test.go

```diff
@@ -162,6 +166,105 @@ func TestOpenSeaProxyApproval(t *testing.T) {
 	}
 }
 
+func TestOpenSeaProxyPreApproval(t *testing.T) {
+
+	tests := []struct {
+		name            string
+		hasProxy        bool
+		wantPreApproved bool
+	}{
+		{
+			name:            "without proxy on mint",
+			hasProxy:        false,
+			wantPreApproved: false,
+		},
+		{
+			name:            "with proxy on mint",
+			hasProxy:        true,
+			wantPreApproved: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sim, nft := deploy(t)
+
+			if tt.hasProxy {
+				openseatest.SetProxyTB(t, sim, sim.Addr(tokenMinter), sim.Addr(proxy))
+			}
+
+			approvalEvents := make(chan *TestableERC721CommonEnumerableApprovalForAll)
```

Comment:
It's much cleaner to use `FilterApprovalForAll()` instead of `WatchApprovalForAll()`. I think I may have introduced the channel + `Watch` pattern elsewhere, before I knew about `Filter`. This allows you to collect _all_ events into a `[]` slice before using `cmp.Diff`; the benefit is that it captures any extras without the `time.After()` hack and no need for the `expected` flag is you literally just test for a known (possibly empty) slice of events.